src/lib/import_export/csv/csv_converter.hpp

```diff
@@ -59,6 +59,11 @@ class CsvConverter : public BaseCsvConverter {
       Assert(_config.null_handling != NullHandling::RejectNullStrings,
              "Unquoted null found in CSV file. Quote it for string literal \"null\", leave field empty for null "
              "value, or set 'null_handling' to the appropriate strategy in parse config.");
+      if (_config.null_handling == NullHandling::NullStringAsNull) {
+        Assert(_is_nullable,
+               "Unquoted null found in CSV file, while the associated column is not nullable. Quote it for string "
+               "literal \"null\".");
+      }
```

Comment:
Could we merge that with below's `if (_is_nullable && _config.null_handling == NullHandling::NullStringAsNull) {`? Isn't `if (_config.null_handling == NullHandling::NullStringAsNull) {` and `Assert(_is_nullable` effectively the same? If that is correct, it might even communicate the intent clearer:
- if we encountered a NULL_STRING
- and if we should treat a NullStringAsNull
- then we assert that the column really `_is_nullable`
- if all of that is the case, we can set `_null_values` to `true` at `position`

or am I wrong?

Reply:
Good point 🙃 . I merged the two cases into one.